microshift.sh

```diff
@@ -0,0 +1,134 @@
+
+# This requirement is taken from https://github.com/openshift/microshift/blob/main/scripts/image-builder/configure.sh
+# Also https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/composing_a_customized_rhel_system_image/installing-composer_composing-a-customized-rhel-system-image
+# list out the dependencies and usecase.
+function configure_host {
+    sudo dnf install -y git osbuild-composer composer-cli ostree rpm-ostree \
+      cockpit-composer cockpit-machines bash-completion podman genisoimage \
+      createrepo yum-utils selinux-policy-devel jq wget lorax
+    sudo systemctl enable osbuild-composer.socket --now
+    sudo systemctl enable cockpit.socket --now
+    sudo firewall-cmd --add-service=cockpit
+    # The home dir and files must have read permissions to group
+    # and others because osbuilder is running from another non-priviledged user account
+    # and allow it to read the files on current user home, it is required.
+    # https://github.com/openshift/microshift/blob/main/scripts/image-builder/configure.sh#L29-L32
+    sudo chmod 0755 $HOME
+}
+
+function enable_repos {
+    sudo subscription-manager repos \
+       --enable rhocp-4.12-for-rhel-8-$(uname -i)-rpms \
+       --enable fast-datapath-for-rhel-8-$(uname -i)-rpms
+}
+
+function download_microshift_rpm {
+    local pkgDir=$1
+    pushd ${pkgDir}
+    brew download-build --quiet --arch noarch --arch ${ARCH} ${MICROSHIFT_NVR}
+    popd
+}
+
+function create_iso {
+    local pkgDir=$1
+    rm -fr microshift
+    git clone -b release-4.12 https://github.com/openshift/microshift.git
+    pushd microshift
+    sed -i '/# customizations/,$d' scripts/image-builder/config/blueprint_v0.0.1.toml
+    cat << EOF >> scripts/image-builder/config/blueprint_v0.0.1.toml
+[[packages]]
+name = "microshift-release-info"
+version = "*"
+EOF
+    sed -i 's/redhat/core/g' scripts/image-builder/config/kickstart.ks.template
+    sed -i '/--bootproto=dhcp/a\network  --hostname=api.crc.testing' scripts/image-builder/config/kickstart.ks.template
+    scripts/image-builder/cleanup.sh -full
+    scripts/image-builder/build.sh -microshift_rpms ${pkgDir} -pull_secret_file ${OPENSHIFT_PULL_SECRET_PATH} -authorized_keys_file $(realpath ../id_ecdsa_crc.pub)
+    popd
+}
+
+enable_repos
+configure_host
+microshift_pkg_dir=$(mktemp -p /tmp -d tmp-rpmXXX)
+# This directory contains the microshift rpm  passed to osbuilder, worker for osbuilder
+# running as non-priviledged user and this tmp directory have 0700 permission. To allow
+# worker to read/execute this file we need to change the permission to 0755
+chmod 0755 ${microshift_pkg_dir}
+download_microshift_rpm ${microshift_pkg_dir}
+create_iso ${microshift_pkg_dir}
+OPENSHIFT_RELEASE_VERSION=$(rpm -qp  --qf '%{VERSION}' ${microshift_pkg_dir}/microshift-4.*.rpm)
+rm -fr ${microshift_pkg_dir}
+sudo cp -Z microshift/_output/image-builder/microshift-installer-*.iso /var/lib/libvirt/images/microshift-installer.iso
```

Comment:
If you move this right after `create_iso`, things are grouped in more natural order, we create the iso, copy it, we get the microshift version and use it to download `oc`.